### ISSUE TITLE
Crash when trying to open a previous repo that no longer exists

### ISSUE THREAD

**Issue by halostatue** [jesseduffield/lazygit#1609]
**Describe the bug**

I answered "no" to initializing a repo and got a crash with instructions to file a bug. The bug appears to be because of `recentrepos[0]` not existing in `state.yml`.

**To Reproduce**

```
~/path/x $ lazygit
Not in a git repository. Create a new git repository? (y/n): n
2021/12/02 10:36:19 An error occurred! Please create an issue at: https://github.com/jesseduffield/lazygit/issues

*fs.PathError chdir /Users/halostatue/path/y/z2: no such file or directory
/private/tmp/lazygit-20211122-54441-55gtnq/lazygit-0.31.4/main.go:141 (0x1030717c4)
/opt/homebrew/Cellar/go/1.17.2/libexec/src/runtime/proc.go:255 (0x102b20b64)
	main: fn()
/opt/homebrew/Cellar/go/1.17.2/libexec/src/runtime/asm_arm64.s:1133 (0x102b511e4)
	goexit: MOVD	R0, R0	// NOP
```

**Expected behavior**

No exception.

**Desktop (please complete the following information):**

 - OS: macOS 11
 - `lazygit --version`: `commit=, build date=, build source=homebrew, version=0.31.4, os=darwin, arch=arm64`
 - Fish shell 3.3.1

**Additional context**

I was in `~/path/x`. I have a folder `~/path/y/z` and I have _in the past_ had a folder `~/path/y/z2`. My `~/Library/Application Support/lazygit/state.yml` contains:

```yaml
lastupdatecheck: 0
recentrepos:
- /Users/halostatue/path/y/z2
startuppopupversion: 0
```

**Comment by strboul** [comment]
A similar bug occurs when providing a path that's not a git directory:

```sh
lazygit -p ~/Documents
# An error occurred! Please create an issue at: https://github.com/jesseduffield/lazygit/issues
#
# *fs.PathError stat /home/user/Documents/.git: no such file or directory
# github.com/jesseduffield/lazygit/pkg/utils/errors.go:13 (0x5570a035fafa)
# github.com/jesseduffield/lazygit/pkg/commands/git.go:159 (0x5570a035fafb)
# github.com/jesseduffield/lazygit/pkg/commands/git.go:62 (0x5570a035e867)
# github.com/jesseduffield/lazygit/pkg/gui/gui.go:254 (0x5570a0412bbc)
# github.com/jesseduffield/lazygit/pkg/gui/gui.go:589 (0x5570a04154b8)
# github.com/jesseduffield/lazygit/pkg/gui/gui.go:625 (0x5570a0415c05)
# github.com/jesseduffield/lazygit/pkg/utils/utils.go:115 (0x5570a02ff407)
# github.com/jesseduffield/lazygit/pkg/gui/gui.go:624 (0x5570a0415b65)
# github.com/jesseduffield/lazygit/pkg/app/app.go:217 (0x5570a0445692)
# github.com/jesseduffield/lazygit/pkg/app/app.go:217 (0x5570a0445672)
# github.com/jesseduffield/lazygit/main.go:154 (0x5570a0453190)
# runtime/proc.go:250 (0x55709fe407d3)
# runtime/asm_amd64.s:1571 (0x55709fe6e261)
```

**Comment by mark2185** [comment]
@strboul I opened up a new issue #2091 with your report, the original issue is solved in `v0.35`.
